bots/task/__init__.py

```diff
@@ -268,7 +269,26 @@ def run(context, function, **kwargs):
         finish(publishing, ret, name, context, issue)
     return ret or 0
 
-def stale(days, pathspec):
+def checkout(ref="HEAD"):
+    if ref:
+        execute("git", "checkout", "--detach", ref)
+
+    # COMPAT: If the bots directory doesn't exist in this branch, check it out from master
+    if subprocess.call([ "git", "ls-tree", "-d", "HEAD:bots/"], stdout=DEVNULL) != 0:
+        sys.stderr.write("Checking out bots directory from master ...\n")
+        subprocess.check_call([ "git", "checkout", "--force", "origin/master", "--", "bots/" ])
+
+        # The machine code is special copy it from master too
```

Comment:
nitpick fix during merge:
`# The machine code is special, copy it from master too`